### ISSUE TITLE
Initial TypeScript type definitions

### ISSUE THREAD

**Issue by programbo** [storyblok/storyblok-js-client#3]
I knocked up a typedef for `storyblok-js-client` so my team has a better experience when they start working on my project.

Some parts were rushed and others are based on my assumptions that might be wrong. I'd really like some feedback on making this more comprehensive 😄 

```typescript
declare global {
  interface StoryblokBridgeConfig {
    initOnlyOnce?: boolean
  }
  interface StoryblokBridge {
    init: (config: StoryblokBridgeConfig) => void
    on: (event: string, callback: () => void) => void
  }
  interface Window {
    storyblok: StoryblokBridge
    StoryblokCacheVersion: number
  }
}

import { AxiosInstance } from 'axios'
interface StoryblokConfig {
  accessToken: string
  cache?: StoryblokCache
  timeout?: number
  headers?: any
}
interface StoryblokCache {
  type?: 'memory'
  clear?: 'auto' | 'manual'
}
export interface Stories {
  data: {
    stories: {
      alternates: string[]
      content: {
        [index: string]: string
      }
      created_at: string
      full_slug: string
      group_id: string
      id: number
      is_startpage: boolean
      meta_data: any
      name: string
      parent_id: number
      position: number
      published_at: string | null
      slug: string
      sort_by_date: string | null
      tag_list: string[]
      uuid: string
    }
  }
}
interface Params {
  token: string
  with_tag?: string
  is_startpage?: 0 | 1
  starts_with?: string
  by_uuids?: string
  excluding_ids?: string
  excluding_fields?: string
  version?: 'draft' | 'published'
  cv?: number
  sort_by?: string
  search_term?: string
  filter_query?: string
  per_page?: number
  page?: string
}
declare class Storyblok {
  throttle: any
  cacheVersion: number
  accessToken: string
  cache: StoryblokCache
  client: AxiosInstance
  constructor(config: StoryblokConfig, endpoint?: string)
  get(slug: string, params: Params): Promise<Stories>
  getToken(): string
  cacheResponse(url: string, params: Params): Promise<Stories>
  newVersion(): number
  cacheProvider(): {
    get(key: string): any
    set(key: string, content: string): void
    flush(): void
  }
  flushCache(): this
}
export default Storyblok
```

**Comment by onefriendaday** [comment]
Thank's!

I've create a new feature branch for typescript support and already adapted the ts file a bit:
https://github.com/storyblok/storyblok-js-client/blob/feature/typescript-support/dist/index.d.ts

I think there needs to be added a few additional get methods to bring types to the general response object as currently "client.get" can return any response object.

Additions like this would let you define types for each response.
https://github.com/storyblok/storyblok-js-client/commit/1b8102e3182c52d29a4d3a75f789643fd76b6596#diff-c60c67bd505d9618ebf77a7f130903d0R51

**Comment by programbo** [comment]
Thanks for this, your changes just taught me more about the API 👍 

I've commented with a couple non-essential additions.

**Comment by onefriendaday** [comment]
Thank you for the comments and suggestions! 

The first version with Typescript definitions has been released "1.0.11". 🎉

**Comment by programbo** [comment]
Great news! Should we do the same thing for `storyblok-react`?

**Comment by programbo** [comment]
Three issues I have come across with the current typings and `storyblok-js-client` v1.0.12:

```typescript
interface StoryblokBridgeConfig {
  initOnlyOnce?: boolean,
  accessToken: string // Not included when calling `window.storyblok.init({ initOnlyOnce: true })`
}
```

```typescript
  interface Window {
    storyblok: StoryblokBridge
    StoryblokCacheVersion: number // Included in the tutorial https://www.storyblok.com/tp/next-js-react-guide
  }
```

```typescript
content: {
  component: string
  _uid: string
  [index: string]: any // Was previously `string` which causes issue when a `Blocks` are added to the schema
}

**Comment by onefriendaday** [comment]
Thanks! Fixed in 1.0.13
